Backpack/StarRating/Classes/BPKStar.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/**
+ * Enum values for specifying star state
+ */
+typedef NS_ENUM(NSInteger, BPKStarState) {
+    BPKStarStateDefault,
+    BPKStarStateHalf,
+    BPKStarStateFull
+};
+
+/**
+ * Enum values for specifying star size
+ */
+typedef NS_ENUM(NSUInteger, BPKStarSize) {
+    BPKStarSizeSmall,
+    BPKStarSizeLarge
+};
+
+NS_ASSUME_NONNULL_BEGIN
+
+NS_SWIFT_NAME(Star) @interface BPKStar : UIView
+
+/**
+ * State of the star
+ * BPKStarStateDefault will render a star filled with starColor
+ * BPKStarStateHalf will render a star filled with half starFilledColor and half starColor
+ * BPKStarStateFull will render a star filled with starFilledColor
+ *
+ * @see BPKStarState
+ */
+@property (nonatomic) BPKStarState state;
+
+/**
+ * Size of the star
+ *
+ * @see BPKStarSize
+ */
+@property(nonatomic) BPKStarSize size;
+
+@property (nullable, nonatomic, strong) UIColor* starColor UI_APPEARANCE_SELECTOR;
+@property (nullable, nonatomic, strong) UIColor* starFilledColor UI_APPEARANCE_SELECTOR;
+
+- (instancetype)initWithSize:(BPKStarSize)size NS_DESIGNATED_INITIALIZER;
```

Comment:
This needs to be documented. In general can we please mention the default values for the different initializers?

Reply:
Done